backend/app/requests/caching.py

```diff
@@ -0,0 +1,81 @@
+"""Methods to cache adequacy and representative points responses."""
+import json
+import logging
+import os
+import six
+
+import ujson
+
+from backend.config import config
+
+logger = logging.getLogger(__name__)
+
+CACHE_DIRECTORY = config.get('cached_result_directory')
+
+
+def cache(hint_fields):
+    """
+    Decorator function for caching API responses.
+
+    If caching is disabled, responses are not cached.
+    If any of the hint fields are empty or missing, responses are not cached.
+
+    Otherwise, attempt to return previously cached responses, saving the responses
+    whose files do not yet exist.
+    """
+    def wrap(f):
+        @six.wraps(f)
+        def wrapped_f(**kwargs):
+            return _cache(func=f, hint_fields=hint_fields, **kwargs)
+        return wrapped_f
+    return wrap
+
+
+def _cache(func, hint_fields, **kwargs):
+    hint_values = [kwargs.get(field) for field in hint_fields]
+    cache_filepath = _get_cached_filepath(
+        prefix=func.__name__,
+        hint_values=hint_values,
+    )
+    # If caching is disabled or a hint is missing, call the function normally.
+    if not all(hint_values) or not config.get('cache_requests'):
+        response = func(**kwargs)
+    # If the file exists, read and return.
+    elif os.path.isfile(cache_filepath):
+        with open(cache_filepath, 'r') as f:
+            response = ujson.load(f)
+        logger.debug('Returning cached response.')
+    # If the file does not exist, calculate and write to the cache.
+    else:
+        response = func(**kwargs)
+        logger.debug('Storing cached response.')
+        with open(cache_filepath, 'w+') as f:
+            json.dump(obj=response, fp=f)
+
+    return response
+
+
+def _get_cached_filepath(prefix, hint_values):
+    """Return the filepath where a cached response would live for the given inputs."""
+    filename = '{prefix}_{hash_value}.json'.format(
+        prefix=prefix,
+        hash_value=_hash_hint_values(hint_values),
+    )
+    return os.path.join(CACHE_DIRECTORY, filename)
+
+
+def _hash_hint_values(hint_values):
```

Comment:
Maybe we should use `hashlib` here? The current hash values are numbers on the order of `1598481923345719655`.

Reply:
You mean take it service_areas + providers instead of hint?